### ISSUE TITLE
History: Prevent polluting history

### ISSUE THREAD

**Issue by miina** [swissspidy/web-stories-wp#10009]
## Feature Description
Currently, history is polluted for example by props that are added async in the background. Such props are for example:
```
  'resource.baseColor',
  'resource.isMuted',
  'resource.poster',
  'resource.posterId',
```

There might be more. In #9822, logic is put in place to allow excluding properties from creating separate history entries.
Once we have both that and #9577 merged, we should have everything in place to not pollute history anymore.

**Comment by spacedmonkey** [comment]
I would also add 
`'resource.blurHash'`
`'resource.id'`

**Comment by swissspidy** [comment]
Let's create a separate ticket for adding all of these properties

**Comment by miina** [comment]
> Let's create a separate ticket for adding all of these properties

@swissspidy This actually is a separate issue for adding the properties. Currently, no properties have been excluded yet from history, only the logic is in place for that (#9645)

Or are you thinking adding one more separate issue for `blurHash` and `id` specifically?

**Comment by swissspidy** [comment]
Oh, didn't realize that. Never mind then!

As for the list of properties, seems like ignoring `isMuted` and `poster` would break manually muting a video or changing its poster.

`resource.baseColor`, `resource.posterId` and `resource.id` seem reasonable at first glance.

As for `resource.blurHash` isn't that only added for media library items? So I don't think it's actually relevant here.

Regardless, this needs careful testing.

**Comment by miina** [comment]
Adding these values to be ignored means that the values will be ignored only if _only_ that value changes, if the change is included in a larger change, the value will still be saved. E.g. if only `isMuted` changes, it won't cause a separate history entry, however, if `isMuted` is included within other changes, it will still be saved.

Some testing showed that adding the following to the ignore list don't seem to cause issues:
```
  'resource.baseColor',
  'resource.id',
  'resource.isMuted',
  'resource.posterId',
  'resource.poster,'
```

If I understand correctly then `resource.poster` does not change when manually changing the poster -- `element.poster` will be change in that case, `element.resource.poster` is always the original poster. Thus it should be ok to ignore this value changing causing a new history entry. LMK if that's incorrect.

I'll do some more testing tomorrow.
With these ignored values, two clicks of "Undo" are needed to undo uploading a new video to the Page (as opposed to 5 on `main`), will try to track down what's still causing the extra step.

**Comment by swissspidy** [comment]
> Adding these values to be ignored means that the values will be ignored only if only that value changes, if the change is included in a larger change, the value will still be saved.

Oh neat! I didn't realize that. That makes things a lot easier then!

**Comment by felipebochehin87** [comment]
Tested against https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin, using PR #10240.

1. Opened https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin
2. Dragged and dropped a video (from local to the story editor)
3. Clicked on Undo - video was removed after the 1st click, whereas on main I needed to click on Undo 5 times
4. Dragged and dropped an image (from local to the story editor)
5. Clicked on Undo - image was removed after the 1st click, whereas on main I needed to click on Undo 5 times
6. Repeated steps above for other actions and elements, such as text, formats, trim videos, ...
